applybn/feature_selection/bn_nmi_feature_selector.py

```diff
@@ -0,0 +1,130 @@
+
+from pyitlib import discrete_random_variable as drv
+import pandas as pd
+
+
+class NMIFeatureSelectior:
+  def __init__(self, threshhold = 0.5, bin_count=100, verboose=False):
+    self.threshhold = threshhold
+    self.bin_count = bin_count
+    self.verboose = verboose
+
+
+  def _discreticise(self, x, y=None):
+    '''
+    Prep data to feed into pyitlib drv.entropy
+    Calc happenings of each event in descreticised feature
+    '''
+    if y is None:
+      discr = [0] * (max(x) - min(x) + 1)
```

Comment:
Consider what happens if columns are empty, or if max(x) == min(x). This would create a list of length 1 in _discreticise, which may cause edge cases.